js/zoom-vanilla.js

```diff
@@ -245,6 +249,10 @@
 
 			document.body.classList.remove('zoom-overlay-transitioning')
 		}
+		// find maximum z-index in document
+		function findMaximumZindex () {
+			return Math.max(1,...Array.from(document.querySelectorAll('*')).map(el => getComputedStyle(el).zIndex).filter(v => !isNaN(parseInt(v[1]))).map(o => parseInt(o)))
```

Comment:
This is going to be an extremely expensive process for the browser to run each time a zoom is requested. Doing it once on initialisation would help, but I wouldn't advise walking the entire DOM and compute the style of every single element.

An options object for the library may be a better solution—though a larger change. Also using, if supported, the `dialog` element which, iirc, will always be on top.

Reply:
Hi @philiprenich 

If I update to this 

```
<!-- if have zIndex then will set the zIndex, If not it will set the default value -->
<img src="img/blog_post_featured.png" data-action="zoom" data-zIndex="2000">
```

It will be better or not?

Reply:
Hi folks, thanks for looking into this. I'm not sure I understand the solution fully in relation to the problem. If the issue is the z-index of `.zoom-img-wrap` being set to 666, this is done in the CSS file and can be easily overridden with a CSS declaration placed anywhere after this CSS file. But perhaps I'm missing something?

Reply:
Sorry for adding churn to the discussion. I hadn't actually looked at how the z-index was set up, just the solution implementation here. @spinningarrow I believe your assessment is correct and overriding the z-index should be correct and trivial to do in the site author's CSS.

Reply:
No worries - thank you for looking into it!